_data/gsoc-ideas.yml

```
@@ -30,13 +30,77 @@
   difficulty: Medium.
   length: long
-  mentors: [antoniojimeneznieto, djspiewak, armanbilge]
+  mentors: [antoniojimeneznieto, djspiewak, mpilquist, armanbilge]
+  categories: [operating systems, programming languages]
+
+- title: Polling-based I/O in FS2
+  description:
+    Cats Effect v3.6.0 introduced the ability to poll for I/O readiness. We want to use polling to reimplement several I/O APIs in FS2, including datagrams, unix sockets, and processes, on the JVM and Native platforms.
+  prereqs: Scala, ability to read C
+  difficulty: Medium.
+  length: long
+  mentors: [armanbilge, mpilquist, antoniojimeneznieto]
+  categories: [operating systems, programming languages]
+
+- title: FS2 Connection API
+  description:
+    TCP-based protocols are common (e.g. HTTP, Postgres, Redis) and are implemented by clients to interface with these services (e.g. Ember, Skunk, Rediculous). The goal of this project is to create a "connection" API that supports pooling, error conditions, and metrics and can be shared by all of our client libraries.
+  prereqs: Scala, ideally some knowledge of networking
+  difficulty: Hard.
+  length: long
+  mentors: [mpilquist, armanbilge]
   categories: [operating systems, programming languages]
 
-- title: "Streaming implementation of an XQuery subset"
-  description: "XQuery allows for transforming XML data in a declarative way. Based on the generic streaming query engine added into the [fs2-data](https://fs2-data.gnieh.org) library, write a frontend to implement a streamable subset of XQuery."
-  prereqs: Scala, ideally some experience with XQuery
-  difficulty: Hard. 
+- title: Web Components for Calico
+  description:
+    Calico is a reactive UI library built with Cats Effect and FS2. Web Components are a standard for creating framework-agnostic, reusable UI elements. The goal of this project is to enable Calico users to access the vast array of web components available by improving its DSL and code-generation.
+  prereqs: Scala, ideally experience with Web APIs
+  difficulty: Medium.
+  length: long
+  mentors: [armanbilge]
+  categories: [web, programming languages]
+
+- title: Upgrade sbt-typelevel to sbt 2
+  description:
+    sbt-typelevel is a plugin for sbt, the Scala build tool, used by hundreds of open source and enterprise projects. sbt 2 is in the final stages of development. We want to upgrade sbt-typelevel to sbt 2 and adopt its new features, such as "project matrix" for cross-building.
+  prereqs: Scala
+  difficulty: Medium.
+  length: long
+  mentors: [mzuehlke, armanbilge]
+  categories: [development tools]
+
+- title: Refresh Davenverse projects
+  description:
+    The Davenverse is a collection of several popular Typelevel libraries, including Mules and Rediculous. Unfortunately, we have fallen behind on their maintenance. We want to move these libraries under the Typelevel org, refresh their build tooling, and bring them up-to-date to ensure their longevity.
```

```suggestion
    The Davenverse is a collection of several popular Typelevel libraries, including Mules and cats-scalacheck. Unfortunately, we have fallen behind on their maintenance. We want to move these libraries under the Typelevel org, refresh their build tooling, and bring them up-to-date to ensure their longevity.
```

Rediculous is actually further down the list of things to migrate, and cats-scalacheck has a surprising number of monthly downloads (more than Mules even, if I'm remembering correctly!)